### ISSUE TITLE
Training get stuck after the first epoch ??!!!

### ISSUE THREAD

**Issue by scstu** [fizyr/keras-retinanet#994]
I have started training Retina Net on my database. I used the default configuration. the first epoch finished correctly but then it stucked in epoch 2.. I tried restarting the training process many times but it couldn't pass the second epoch 

I am training on a dataset with approximately 22k images, ~34k instance with only one class .. my validation set contains around 7k instance..

### To start Training I used: 
`!keras_retinanet/bin/train.py --epochs 100 --tensorboard-dir ./logs --snapshot-path ./snapshots --weights snapshots/resnet50_coco_best_v2.1.0.h5 csv train_annotations.csv labels.csv --val-annotations val_annotations_.csv`

### The training stuck point:
```
Epoch 1/100
10000/10000 [==============================] - 3137s 314ms/step - loss: 1.3669 - regression_loss: 1.0304 - classification_loss: 0.3365
Running network: 100% (3145 of 3145) |###| Elapsed Time: 0:06:32 Time:  0:06:32
Parsing annotations: 100% (3145 of 3145) || Elapsed Time: 0:00:00 Time:  0:00:00
6499 instances of class hand with average precision: 0.8951
mAP: 0.8951

Epoch 00001: saving model to ./snapshots/resnet50_csv_01.h5
Epoch 2/100
 7132/10000 [====================>.........] - ETA: 14:36 - loss: 1.2182 - regression_loss: 0.9201 - classification_loss: 0.2981
```

### nvidia-smi output 
```
nvidia-smi
Thu May  9 01:06:47 2019       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 384.90                 Driver Version: 384.90                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla P100-PCIE...  Off  | 00000000:02:00.0 Off |                    0 |
| N/A   30C    P0    29W / 250W |   8703MiB / 12193MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla P100-PCIE...  Off  | 00000000:03:00.0 Off |                    0 |
| N/A   34C    P0    31W / 250W |    353MiB / 12193MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0     10851      C   python                                      8693MiB |
|    1     10851      C   python                                       343MiB |
+-----------------------------------------------------------------------------+
```

Please any help or suggestion to fix that, it will be appreciated.

**Comment by scstu** [comment]
starting the training again with the following modification: 

1. I don't use --log 
2. Changing Number of steps to 5000

but again got stuck at: 

`
Epoch 4/100
2132/5000 [===========>..................] - ETA: 14:10 - loss: 1.1891 - regression_loss: 0.9010 - classification_loss: 0.2881
`
**it seems that it's the same point as previous since I have change the number of the steps 
so after step 17132, it stops !!!**
Please, any Help!!

**Comment by LittleCherry11** [comment]
maybe you could disable the multiprocessing choice by setting --workers 0.

**Comment by scstu** [comment]
I came here to say that @LittleCherry11  thank you indeed! 
The solution to this issue is adding 
`--workers 0 `

I am closing this issue because it's already solved.
